stacks-common/src/util/lru_cache.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2024 Stacks Open Internet Foundation
```

Comment:
Should we consider an LRU cache crate like [this](https://crates.io/crates/lru) one rather than write our own?

Reply:
I considered, but this crate uses a lot of `unsafe`, so I wrote this simple version instead.